File: prime.py
import sys

def sieve_of_eratosthenes(max_prime):
    primes = [True] * (max_prime + 1)
    primes[0] = primes[1] = False
    for i in range(2, int(max_prime**0.5) + 1):
        if primes[i]:
            for j in range(i*i, max_prime + 1, i):
                primes[j] = False
    return [i for i, prime in enumerate(primes) if prime]

def main():
    if len(sys.argv) != 3:
        print("Usage: python prime_calculator.py <max_prime> <num_iterations>")
        sys.exit(1)

    max_prime = int(sys.argv[1])
    num_iterations = int(sys.argv[2])

    for _ in range(num_iterations):
        primes = sieve_of_eratosthenes(max_prime)
        print(f"Found {len(primes)} primes up to {max_prime}")

if __name__ == "__main__":
    main()

File: prime.js
function sieveOfEratosthenes(maxPrime) {
    const primes = new Array(maxPrime + 1).fill(true);
    primes[0] = primes[1] = false;

    for (let i = 2; i <= Math.sqrt(maxPrime); i++) {
        if (primes[i]) {
            for (let j = i * i; j <= maxPrime; j += i) {
                primes[j] = false;
            }
        }
    }

    return primes.map((prime, index) => prime ? index : null).filter((val) => val !== null);
}

function main() {
    const [,, maxPrime, numIterations] = process.argv;

    if (!maxPrime || !numIterations) {
        console.error("Usage: node prime_calculator.js <max_prime> <num_iterations>");
        process.exit(1);
    }

    for (let i = 0; i < numIterations; i++) {
        const primes = sieveOfEratosthenes(Number(maxPrime));
        console.log(`Found ${primes.length} primes up to ${maxPrime}`);
    }
}

main();

